src/services/github.ts

```diff
@@ -314,16 +344,14 @@ export async function githubUpload(context: ContextType<typeof AppContext>, base
         existingFigures = [];
     }
 
-    const figurePaths: string[] = existingFigures.map((f: { path: string; }) => f.path);
-    let i = 0;
-    let proposedName, proposedPath;
-    do {
-        proposedName = name.substring(0, name.lastIndexOf(".")) + ( i ? "_" + (i+1) : "") + name.substring(name.lastIndexOf("."));
-        proposedPath = figurePath + "/" + proposedName;
-        i++;
-    } while(figurePaths.includes(proposedPath))
-
-    const result = await githubCreate(context,  figurePath, proposedName, content);
+    const figurePaths: { path: string, sha: string }[] = existingFigures.map((f: { path: string, sha: string; }) => ({ path: f.path, sha: f.sha }));
+    const figureToReplace = figurePaths.find(f => f.path === figurePath + '/' + name);
+    let result;
+    if (isDefined(figureToReplace)) {
+        result = await githubUpdate(context, figurePath, name, content, figureToReplace.sha);
+    } else {
+        result = await githubCreate(context, figurePath, name, content);
+    }
 
     return `figures/${result.content.name}`;
```

Comment:
The UI of this feature is currently really quite opaque. On replacement there is no indication that anything has happened and the image in the editor doesn't even change when it has changed in GitHub.